feat(dropdown): maintain a lower opacity of the clear icon unless being hovered

ko2in opened the pull request:
## Description
When the dropdown allows for clearing, the clear icon should maintain a lower opacity unless it's being hovered. This improves user experience by indicating the option to clear the selection only when the user interacts with it.

## Testcase
**Current Version:** https://fomantic-ui.com/modules/dropdown.html#clearable-selection

Click the dropdown menu and check the clear icon. It displays with full opacity.

**New Version:** https://jsfiddle.net/ko2in/6gerqn2c/

Click the dropdown menu and check the clear icon. It will remain visible with low opacity until you hover the mouse over the icon.

## Screenshot
**Current Version:**
![Dropdown-Old-Feature](https://github.com/user-attachments/assets/a2d20dd8-e43c-4611-b703-d5e7ef613ba1)


**New Version:**
![Dropdown-New-Feature](https://github.com/user-attachments/assets/8be614f1-7bcd-456a-9530-56f51fbac0a7)


## Closes
#3211

ko2in committed:
feat(dropdown): maintain a lower opacity of the clear icon unless being hovered

When the dropdown allows for clearing, the clear icon should maintain
a lower opacity unless it's being hovered. This improves user experience
by indicating the option to clear the selection only when the user interacts
with it.

lubber-de commented on src/definitions/modules/dropdown.less:
```diff
@@ -616,13 +616,17 @@ select.ui.dropdown {
-    .ui.ui.selection.dropdown:focus > i.icon {
-        opacity: @selectionIconFocusOpacity;
+    .ui.ui.selection.dropdown {
+        &:focus, &:has(> input:focus) {
```
The `:has` selector is [too new](https://caniuse.com/?search=%3Ahas) (unless we decide to drop every browser older than May 2022 (iOS 15.4) or even December 2023(!) when Firefox activated this by default).... (Even the decision to drop older than 2021 wasn't clear #2854)
So the selector should be wrapped as we always do whenever we need to use :has to make things work:

```less
@supports selector(:has(.f)) {
....
}
```

The previous selector is already wrapped into that, so you might just move your additional selector into the support block as well

ko2in replied:
@lubber-de Updated. Thanks.

ko2in committed:
feat(dropdown): wrapped with a @supports selector to ensure CSS pseudo selector :has doesn't break on unsupported devices

ko2in committed:
feat(dropdown): fix eslint error

mvorisek commented:
#3211 is about clickable icons in general. It would be great if there can be some "middle opacity" (opacity after field is focused, but icon not hovered), as there can be more clickable icons which UX will be inconsistent with the "remove icon" after this PR.

ko2in commented:
@mvorisek I don't see there will be more clickable icons except the current arrow and clear icons as this component is not typically designed to include multiple actionable elements in accordance with its intended user experience.

Even so, it's not possible to know what icons should have lower opacity than other icon unless we explicitly specify. So, we still need to extend the current CSS in this case.

The default opacity for the icon in it's non-interacting state is **0.5**. Any adjustment between **0.5** and **1** for such a tiny icon, result in a change that is likely to be barely noticeable. So, I don't see any significant improvement here.

mvorisek commented:
Here https://jsfiddle.net/sq5b1ewL/ is a demo, I am neutral on it as it look slighly more usable on my monitor, but not sure how it renders on other displays.

ko2in committed:
feat(dropdown): additional opacity for action icon for focus state

Display the action icon with the opacity level between inactive and
active state, when the input field in dropdown is focused.

ko2in commented:
I've added an additional state for remove icon with the opacity level between inactive and active state, which is 68% of active state as in @mvorisek example.

I chose to use the variable name `@actionIconFocusOpacity` instead of `@removeIconFocusOpacity` to prevent any potential confusion, as it may represent something other than removal in the context of user customization.

lubber-de commented on src/definitions/modules/dropdown.less:
```diff
@@ -616,13 +616,22 @@ select.ui.dropdown {
+                    opacity: @selectionIconFocusOpacity;
+                }
+                & > i.icon.remove:not(:hover) {
+                    opacity: calc(@actionIconFocusOpacity);
```
The calc method should be either moved to the variable declaration itself or escaped here if your intention is to generate "calc()" as real CSS property.
calc() itself is a LESS function, so in some situations the value is calculated by LESS itself. Don't know your intention, but to allow the user to define anything as opacity we should move the calc() inside the variable.

Here is a short overview about the differences
![image](https://github.com/user-attachments/assets/47723bf2-f7d7-47e6-aa2a-8e19f4fe6248)

ko2in replied:
Already fixed.

ko2in replied:
Sorry. I just remove `calc()` as I think it's unnecessary. We can safely use the evaluated value from LESS.

ko2in committed:
feat(dropdown): explicitly use calc() inside variable

ko2in committed:
feat(dropdown): use only evaluated value without calc()

ko2in commented:
@lubber-de I've rebased and updated again. Can you please merge it before it's too outdated from base branch?

lubber-de commented on src/definitions/modules/dropdown.less:
```diff
@@ -616,13 +616,22 @@ select.ui.dropdown {
             border-color: @selectionFocusBorderColor;
             box-shadow: none;
-            & > i.icon {
+            & > i.icon:not(.remove) {
                 opacity: @selectionIconFocusOpacity;
             }
```
This block can now be removed as it is doubled by your new selector block below. And the check for notInverted doesnt seem to be relevant (for the icons) (your new selector overrides it anyway and the jsfiddle works fine)

ko2in replied:
Thanks. Fixed.

ko2in committed:
remove duplication

lubber-de reviewed (approved):
LGTM
